cmd/posix.go

```diff
@@ -642,6 +648,75 @@ func (s *posix) DeleteVol(volume string) (err error) {
 	return nil
 }
 
+// Walk - is a sorted walker which returns file entries in lexically
+// sorted order, additionally along with metadata about each of those entries.
+func (s *posix) Walk(volume, dirPath, marker string, recursive bool, leafFile string, readMetadataFn func(s StorageAPI, volume, entry, leafFile string) FileInfo) (ch chan FileInfo, err error) {
```

Comment:
since posix is backend aware now, leafFile and readMetadataFn should not be passed

Reply:
I don't agree on that @krishnasrinivas - POSIX is not aware of any such requirement and doesn't need to - it is still satisfying the required separation.

Reply:
that is what we decided @harshavardhana  ... that posix layer is XL-backend aware.

notice that `FileInfo` returned by posix already has info like `ObjectPartInfo` i.e lines are already blurred.

also notice that in this patch storage-rest-server has become XL-aware by passing readMetadata function ... which is not right.

Reply:
They are blurred but I don't want code spilling over all over the place just for the sake of it, it makes sense to keep it simple as is I would do it.  Converting POSIX to be tightly XL aware is another refactor which is not part of this PR @krishnasrinivas and I don't want to hold this one up for something we have to do it more clearly not just one off Walk() function.

Reply:
ok then, we can think about that once this stabilizes.